src/irace/__init__.py

```diff
@@ -149,10 +187,15 @@ def set_initial(self, x):
         
     def run(self):
         """Returns a Pandas DataFrame, one column per parameter and the row index are the configuration ID."""
-        self.scenario['targetRunner'] = self.r_target_runner
+        self.r_target_runner_parallel = make_target_runner_parallel(self.target_aq, self.target_rq, self._pkg.check_output_target_runner, self.scenario.copy())
+        self.scenario['targetRunnerParallel'] = self.r_target_runner_parallel
```

Comment:
Good catch. I thought to just spawn a worker regardless but that might create a problem on platforms where a user's process is limited. I'll change it.

Reply:
In the change, I just made the target runner parallel to call the function directly if `parallel` is 0 or 1. Hope this is ok.

Reply:
If everything is otherwise the same, I would prefer to use `targetRunnerParallel` instead of using `targetRunner` even when user does not use parallel because then I could use pure python function and pure python objects through the queues.

Reply:
I still think that your `targetRunnerParallel` is much more complex than the code it replaces. Is the main benefit to avoid conversions between R/python and increase speed? Also forcing users to use `targetRunnerParallel` means that they cannot use any of the other parallelization mechanisms in irace, such as MPI/qsub and their own `targetRunnerParallel`, they cannot use the automatic retries and maybe other things that I have not even thought about.

Thus, I would rather prefer to add an option to irace.__init__(), for example, "python_parallel=True" that selects to either use your python-based `targetRunnerParallel` or simply uses the R implementation without changing `targetRunnerParallel` as it is now.

Reply:
To be honest, I think it's in part due to me not understanding how to use MPI and not seeing the benefit of it. I started it because I thought it would make it faster, and there was also a problem with garbage collection with the old code which I can't reproduce nor know what's causing it, but I thought spawning new python threads from an embedded R might trigger some edge case in the gc. 

It is possible to avoid conversion between R/python without this `targetRunnerParallel`. But does MPI create new threads for each target runner run, or does it reuse the same thread like a worker pool? If it creates a new thread every time, it might have more overhead than the worker queue that I wrote here. 

The only benefit, it remains, is that it fixes the crash on Windows (#16). I think the proper way to fix it is to fix it in the irace R code / MPI / rpy2. But it seems that the error is due to some low level dynamically linked library / shared symbol. I don't think I have the expertise to understand and fix it. 

> Thus, I would rather prefer to add an option to irace.init(), for example, "python_parallel=True" that selects to either use your python-based targetRunnerParallel or simply uses the R implementation without changing targetRunnerParallel as it is now.

I would actually not prefer this. This creates some confusion for the user. It also create more work for us to maintain two separate logic for stuff like #30. So I would propose that we either close this PR and declare it a failed experiment, or we try to replicate the irace functions like using MPI/qsub, allowing users to set `targetRunnerParallel` and implement retires.